code/game/turfs/floor_types.dm

```diff
@@ -264,6 +265,67 @@
 		for(var/obj/effect/decal/cleanable/C in contents) //for the off chance of someone bleeding mid=flight
 			qdel(C)
 
+/turf/open/floor/almayer/empty/multi_elevator
+	var/is_bottom_of_shaft = FALSE
+
+/turf/open/floor/almayer/empty/multi_elevator/Entered(atom/movable/AM)
+	if(AM == src || istype(AM, /obj/docking_port/stationary/vehicle_elevator) || istype(AM, /obj/structure) ||	istype(AM, /obj/effect/projector) || istype(AM, /obj/effect/landmark/multi_fakezlevel_supply_elevator) || istype(AM, /obj/effect/step_trigger/clone_cleaner) || istype(AM, /obj/effect/elevator/supply/multi) || istype(AM, /atom/movable/clone) || istype(AM, /obj/effect/elevator/animation_overlay))
+		return
+	. = ..()
+
+/turf/open/floor/almayer/empty/multi_elevator/enter_depths(atom/movable/AM)
+	if(AM.throwing == 0 && istype(get_turf(AM), type))
+		//This is assuming the shaft is alined across fake_zlevels -- if you're getting moved to random locations blame the mappers
+		//also that the fake zlevels of the shipmap are 101 tiles apart
+		message_admins("[AM] ---> [AM.type]")
+		var/turf/fallonto_turf = get_turf(src)
+		var/area/my_area = get_area(fallonto_turf)
+		if(supply_controller.shuttle && istype(supply_controller.shuttle, /datum/shuttle/ferry/supply/multi) && my_area.fake_zlevel + 1 <= length(GLOB.supply_elevator_turfs))
+			var/datum/shuttle/ferry/supply/multi/m_shuttle = supply_controller.shuttle
+			if(m_shuttle.moving_status == SHUTTLE_INTRANSIT)
+				for(var/turf/T in m_shuttle.get_location_area())
+					fallonto_turf = T
+					AM.plane = FLOOR_PLANE
+					AM.layer = UNDER_TURF_LAYER
+					break
+			else if(my_area.fake_zlevel + 1 <= length(GLOB.supply_elevator_turfs))
+				for(var/i=0; i<101; i++)
```

Comment:
```suggestion
				for(var/i in 0 to 101)
```
what's this magic number also

Reply:
line 279, I thought about making this number part of the map config... but ya know

Reply:
In next commit I added a proc that finds this number based on searching the map for it, so magic number solved.